Spikes/movement/Character.cpp

```diff
@@ -0,0 +1,143 @@
+#include "stdafx.h"
+#include "Character.h"
+
+
+
+//GLOBAL VARIABLES (FOR TESTING PURPOSES)
+const int PATH_RADIUS = 200;
+const float PI = std::acos(-1);
+
+
+
+Character::Character()
+{
+
+}
+
+Character::Character(float radius, float speed, Color green, 
+					 Color yellow, Vector2f centre) 
+:	m_move(false), 
+	m_centre(centre),
+	m_radius(radius),
+	m_speed(speed),
+	m_angle(90 * PI / 180.0f),
+	m_targetRadius(radius / 5),
+	m_targetAngle(m_angle)
+{
+	m_pos = Vector2f(m_centre.x + PATH_RADIUS * cos(m_angle),
+					 m_centre.y + PATH_RADIUS * sin(m_angle));
+	m_targetPos = m_pos;
+
+	//Character shape
+	m_shape = CircleShape(m_radius);
+	m_shape.setFillColor(green);
+	m_shape.setPosition(Vector2f(m_pos.x - m_radius, m_pos.y - m_radius));
+
+	//Target shape
+	m_target = CircleShape(m_targetRadius);
+	m_target.setFillColor(yellow);
+	m_target.setPosition(Vector2f(m_targetPos.x - m_targetRadius, 
+								  m_targetPos.y - m_targetRadius));
+}
+
+
+void Character::update()
+{
+	resetAngles();	//if above 360 or below 0 degrees
+
```

Comment:
does reset angles need to be called in update or only whenever you activate the pointer?